tensorflow/compiler/tf2tensorrt/convert/ops/resize.cc

```diff
@@ -132,56 +478,13 @@ class ConvertResize : public OpConverterBase<ConvertResize> {
     // Given input shape [N, C, H, W] and output size [H_out, W_out],
     // output shape equals [N, C, H_out, W_out].
     TRT_TensorOrWeights size_input = params_->inputs.at(1);
-    nvinfer1::Dims output_shape_dims;
-    nvinfer1::ITensor* output_shape_tensor{nullptr};
-
-    if (attrs_.static_output_shape) {
-      // If the output shape can be fully determined at build time, calculate it
-      // as a set of dimensions.
-      output_shape_dims.nbDims = input_tensor_->getDimensions().nbDims;
-      for (int i = 0; i < output_shape_dims.nbDims; ++i) {
-        output_shape_dims.d[i] = input_tensor_->getDimensions().d[i];
-      }
-      const int* weights_ptr =
-          params_->inputs.at(1).weights().GetPointer<int>();
-      output_shape_dims.d[output_shape_dims.nbDims - 2] = weights_ptr[0];
-      output_shape_dims.d[output_shape_dims.nbDims - 1] = weights_ptr[1];
-    } else {
-      // Otherwise, build the output shape as a shape tensor that will be
-      // computed at run time. The batch size and num of channels will be copied
-      // from the input shape.
-      StatusOr<nvinfer1::IGatherLayer*> batch_and_chan_dim =
-          builder->GatherDims(input_tensor_->trt_tensor(),
-                              /*indices=*/{0, 1});
-      TRT_ENSURE_PTR_OK(batch_and_chan_dim);
-
-      // The height and width will be obtained from the requested output size.
-      nvinfer1::ITensor* spatial_dims{nullptr};
-      if (size_input.is_weights()) {
-        // If the output size is constant, the height and width dimensions can
-        // be created as constants from the size values.
-        const int* weights_ptr = size_input.weights().GetPointer<int>();
-        StatusOr<nvinfer1::IConstantLayer*> spatial_const =
-            builder->Constant({weights_ptr[0], weights_ptr[1]});
-        TRT_ENSURE_PTR_OK(spatial_const);
-        spatial_dims = (*spatial_const)->getOutput(0);
-      } else {
-        spatial_dims = size_input.tensor()->trt_tensor();
-      }
-
-      StatusOr<nvinfer1::IConcatenationLayer*> result = builder->Concat(
-          {(*batch_and_chan_dim)->getOutput(0), spatial_dims}, 0);
-      TRT_ENSURE_PTR_OK(result);
-      output_shape_tensor = (*result)->getOutput(0);
-    }
 
     ITensorProxyPtr output;
     // For TensorRT < 8, we require a workaround for (bilinear,
     // align_corners=false). Otherwise, create a single TRT IResizeLayer.
     if (!attrs_.align_corners && attrs_.mode == nvinfer1::ResizeMode::kLINEAR &&
         !IS_TRT_VERSION_GE(8, 0, 0, 0)) {
-      StatusOr<nvinfer1::ITensor*> out =
-          ConvertBilinearWorkaround(output_shape_dims, output_shape_tensor);
+      StatusOr<nvinfer1::ITensor*> out = ConvertBilinearWorkaround();
```

Comment:
Well, it may be impossible to test TRT7 on systems with newer GPUs, but we can test the workaround path with TRT>=8 if we allow it to be used via environment variable. I added this in the last commit. So I'd prefer not to do that as we wouldn't be able to the workaround code path on newer systems.

In addition, I tend to think using the preprocessor like that should be avoided. It allows people to make changes that break  the code silently by modifying sections inside `#ifdef` guards that aren't covered by CI systems. So unless there's an API that only exists in version X and thus requires `#ifdef` gaurds, it's safer to avoid that use (IMO). See for example [this PR](https://github.com/tensorflow/tensorflow/pull/51039).